packages/compass-saved-aggregations-queries/src/index.ts

```diff
@@ -1,23 +1,53 @@
-import type AppRegistry from 'hadron-app-registry';
-import store from './stores/index';
-import Component from './components/index';
+import { registerHadronPlugin } from 'hadron-app-registry';
+import { dataServiceLocator } from 'mongodb-data-service/provider';
+import { mongoDBInstanceLocator } from '@mongodb-js/compass-app-stores/provider';
+import { createLoggerAndTelemetryLocator } from '@mongodb-js/compass-logging/provider';
+import type { DataService } from 'mongodb-data-service';
+import { activatePlugin } from './stores';
+import AggregationsQueriesList from './components/aggregations-queries-list';
+import type {
+  PipelineStorage,
+  RecentQueryStorage,
+} from '@mongodb-js/my-queries-storage';
 
-const role = {
-  component: Component,
-  name: 'My Queries',
-  order: 1,
-};
-
-function activate(appRegistry: AppRegistry): void {
-  appRegistry.registerStore('App.AggregationsQueriesListStore', store);
-  appRegistry.registerRole('Instance.Tab', role);
+function activate(): void {
+  // noop
 }
 
-function deactivate(appRegistry: AppRegistry): void {
-  appRegistry.deregisterStore('App.AggregationsQueriesListStore');
-  appRegistry.deregisterRole('Instance.Tab', role);
+function deactivate(): void {
+  // noop
 }
 
-export { activate, deactivate };
+const serviceLocators = {
+  dataService: dataServiceLocator as typeof dataServiceLocator<
+    // Getting passed to the mongodb instance so hard to be more explicit
+    // about used methods
+    keyof DataService
+  >,
```

Comment:
I wonder if we should just pass data service to instance constructor to avoid always passing two services only to use one

Reply:
Well … this sounds a bit like it would make sense to expose more than the `MongoDBInstance` instance from the instance-store provider, and also the methods we had previously semi-exposed around it, like `fetchCollectionDetails` and `fetchDatabaseDetails`? Those seem like the only things we’d actually be using in this package anyway

Reply:
Maybe, yeah! I'm "marking" this for now as something to just think over a bit more / keep in mind, it might be that this is one of the few places that actually needs it, so maybe it's okay to keep as-is 🙂